### ISSUE TITLE
Unexpected `nil` value from module service when using Terragrunt dependencies outside workdir.

### ISSUE THREAD

**Issue by kaorihinata** [leg100/pug#100]
**Describe the bug**

I have a set of modules that interdepend via Terragrunt. When I enter the directory of one module (`aws_vpcs` for example) and run `pug` with `-p terragrunt` (straight `terragrunt` and `tofu` have no issues, but I need outputs), I receive the following backtrace:

```
Caught panic:

runtime error: invalid memory address or nil pointer dereference

Restoring terminal...

goroutine 1 [running]:
runtime/debug.Stack()
	/Users/nn/.sandbox/libexec/go/src/runtime/debug/stack.go:24 +0x64
runtime/debug.PrintStack()
	/Users/nn/.sandbox/libexec/go/src/runtime/debug/stack.go:16 +0x1c
github.com/charmbracelet/bubbletea.(*Program).Run.func1()
	/Users/nn/Projects/sports-src/sysutils/pug/work/tmp_gopath/pkg/mod/github.com/charmbracelet/bubbletea@v0.26.2/tea.go:479 +0x88
panic({0x101498ba0?, 0x101875e30?})
	/Users/nn/.sandbox/libexec/go/src/runtime/panic.go:770 +0x124
github.com/leg100/pug/internal/tui/module.(*ListMaker).Make.func1(0x140004f4b80)
	/Users/nn/Projects/sports-src/sysutils/pug/work/pug-0.4.1/internal/tui/module/list.go:80 +0x454
github.com/leg100/pug/internal/tui/table.(*Model[...]).renderRow(0x0, 0x0?)
	/Users/nn/Projects/sports-src/sysutils/pug/work/pug-0.4.1/internal/tui/table/table.go:652 +0x160
github.com/leg100/pug/internal/tui/table.(*Model[...]).UpdateViewport(0x101531f20)
	/Users/nn/Projects/sports-src/sysutils/pug/work/pug-0.4.1/internal/tui/table/table.go:328 +0xf0
github.com/leg100/pug/internal/tui/table.(*Model[...]).SetItems(0x101531f20, 0x14000268660)
	/Users/nn/Projects/sports-src/sysutils/pug/work/pug-0.4.1/internal/tui/table/table.go:573 +0x52c
github.com/leg100/pug/internal/tui/table.Model[...].Update(_, {_, _})
	/Users/nn/Projects/sports-src/sysutils/pug/work/pug-0.4.1/internal/tui/table/table.go:214 +0xc54
github.com/leg100/pug/internal/tui/module.list.Update({{0x10152ba10, 0x14000264d90}, {0x101529d40, 0x1400025ae40}, {0x101529d80, 0x1400003e780}, {{0x14000218540, 0x5, 0x8}, {0x14000630018, ...}, ...}, ...}, ...)
	/Users/nn/Projects/sports-src/sysutils/pug/work/pug-0.4.1/internal/tui/module/list.go:181 +0x16f0
github.com/leg100/pug/internal/tui.(*Cache).Update(...)
	/Users/nn/Projects/sports-src/sysutils/pug/work/pug-0.4.1/internal/tui/cache.go:45
github.com/leg100/pug/internal/tui.(*Cache).UpdateAll(0x14000064938, {0x1014bfb60, 0x14000630300})
	/Users/nn/Projects/sports-src/sysutils/pug/work/pug-0.4.1/internal/tui/cache.go:38 +0x104
github.com/leg100/pug/internal/tui/top.model.Update({0x1400025b040, {0x10152ba10, 0x14000264d90}, 0x69, 0x1b, 0x0, 0x0, 0x0, 0x0, {0x14000029100, ...}, ...}, ...)
	/Users/nn/Projects/sports-src/sysutils/pug/work/pug-0.4.1/internal/tui/top/model.go:272 +0x176c
github.com/charmbracelet/bubbletea.(*Program).eventLoop(0x14000229110, {0x101527ef8?, 0x1400025eaa0?}, 0x1400002a8a0)
	/Users/nn/Projects/sports-src/sysutils/pug/work/tmp_gopath/pkg/mod/github.com/charmbracelet/bubbletea@v0.26.2/tea.go:412 +0x540
github.com/charmbracelet/bubbletea.(*Program).Run(0x14000229110)
	/Users/nn/Projects/sports-src/sysutils/pug/work/tmp_gopath/pkg/mod/github.com/charmbracelet/bubbletea@v0.26.2/tea.go:547 +0x6e4
github.com/leg100/pug/internal/app.Start({0x101524bb8, 0x14000064028}, {0x101524bb8?, 0x14000064030?}, {0x14000020160?, 0x0?, 0x14000039f38?})
	/Users/nn/Projects/sports-src/sysutils/pug/work/pug-0.4.1/internal/app/app.go:70 +0x320
main.main()
	/Users/nn/Projects/sports-src/sysutils/pug/work/pug-0.4.1/main.go:11 +0x70
```

After adding some `Println`s and `Sleep`s for debugging it turns out that it's the call to `m.ModuleService.Get(id)` in `internal/tui/module/list.go` (in `Make(...)` specifically) that's returning `nil`, and given the lack of error handling around the line I assume it isn't expected to fail.

Another thing that may be relevant is that this only seems to be occurring when Terragrunt has dependencies that pug may not be aware of due to the `workdir`. In this case, the 2 modules I've had this issue with have dependencies that are in the parent directory (so `../aws_iam_service_accounts`, for example.)

Version is 0.4.1, but I tested the 0.4.0 build and it does it as well.

**Setup (please complete the following information):**
 - OS: macOS Sequoia (15.0)
 - Shell: zsh
 - Terminal Emulator: Terminal.app
 - Terminal Multiplexer: None

**To Reproduce**
1. Decompress the attached [test modules](https://github.com/user-attachments/files/16236590/test_modules.zip).
2. `cd test_module_b`
3. `terragrunt run-all apply`
4. `pug -p terragrunt`

**Expected behavior**
Hard to say. I would like for it to follow those dependencies (as terragrunt does), but since everything depends on the workdir I don't know how much this would complicate code. Perhaps a more meaningful error would work instead.

**Comment by leg100** [comment]
I actually designed Pug to ignore external dependencies. But clearly you've found a bug with its implementation, which I've now fixed in #101. The fix also includes the error handling you pointed out was missing. What should happen now is that the external dependency should be ignored and you'll see a warning in the logs. I think the warning could improved to provide further context, but I think this will do for now.

So, yes, while you would like Pug to follow external dependencies, I think it would indeed complicate the design. Is there a reason why you don't run Pug in the parent directory of both/all modules, so that all dependencies are _internal_?

**Comment by leg100** [comment]
Released fix here: https://github.com/leg100/pug/releases/tag/v0.4.2

**Comment by kaorihinata** [comment]
> Is there a reason why you don't run Pug in the parent directory of both/all modules, so that all dependencies are _internal_?

I do run it in the parent. I was testing something else with terragrunt and hadn't backed out to the parent directory. Ran, crashed, and figured "I should probably report it since it is still a crash." I said "Hard to say" specifically because it doesn't make sense to leave the workdir with the way you've designed pug, but it also worked differently than terragrunt's dependency resolution so I provided both.
